### ISSUE TITLE
Expose explanation for a code

### ISSUE THREAD

**Issue by jankatins** [charliermarsh/ruff#4343]
In https://github.com/koxudaxi/ruff-pycharm-plugin/issues/147 we were discussing the possibility to annotate the `noqa: <code>` codes with the example text as it is on the website. Is there a way to get this information from ruff? E.g. something like `ruff explain <code>` would return the explanation for the code (in machine readable format, e.g. `--json`)?

**Comment by JonathanPlasse** [comment]
You can try
```console
ruff rule <code>
```

**Comment by charliermarsh** [comment]
Hmm, yeah, the JSON output should probably include the documentation? Similarly to how `ruff rule S101` shows:

```markdown
# assert (S101)

Derived from the **flake8-bandit** linter.

## What it does
Checks for uses of the `assert` keyword.

## Why is this bad?
Assertions are removed when Python is run with optimization requested
(i.e., when the `-O` flag is present), which is a common practice in
production environments. As such, assertions should not be used for runtime
validation of user input or to enforce  interface constraints.

Consider raising a meaningful error instead of using `assert`.
```

**Comment by JonathanPlasse** [comment]
I can work on it.

**Comment by jankatins** [comment]
BTW: is it intended that the non-json output never shows the summary? `Use of `assert` detected` is in `ruff rule S101 --format=json` but not in `ruff rule S101`. I would have expected that this is basically the headline: `# assert (S101): Use of 'assert' detected` or so.

**Comment by JonathanPlasse** [comment]
I would be in favor of always showing `message_formats`.
